### PR TITLE
(#475) Add support for server and process uptime fields

### LINKED ISSUE #475: Support for server.uptime and process.uptime
Is is worth it adding the milliseconds a server has been up? It's a value coming directly from the OS (both *NIX and Windows) using the `uptime` command.

For process it already exists a `process.start` which could define how to calculate the uptime.

I.e., as described here: (https://superuser.com/a/464413) in linux/mac, it's possible to gather this info:
```
$ ps -eo pid,comm,lstart,etime,time,args
  PID COMM             STARTED                          ELAPSED      TIME ARGS
    1 /sbin/launchd    Sun May 19 10:10:53 2019     09-04:18:37  38:47.10 /sbin/launchd
   39 /usr/sbin/syslog Sun May 19 10:11:01 2019     09-04:18:29   0:18.02 /usr/sbin/syslogd
   40 /usr/libexec/Use Sun May 19 10:11:01 2019     09-04:18:29   0:34.19 /usr/libexec/UserEventAgent (System)
   43 /System/Library/ Sun May 19 10:11:01 2019     09-04:18:29   0:14.74 /System/Library/PrivateFrameworks/Uninstall.framework/Resources/
   44 /usr/libexec/kex Sun May 19 10:11:01 2019     09-04:18:29   0:22.23 /usr/libexec/kextd
```

But again, please close this issue if you do not consider it :)

Thanks!

### PR DESCRIPTION
**Pull request by mdelapenya**
Closes #475

## What does this PR do?
This PR adds fields for `process.uptime` and `server.uptime`. As suggested by @webmat, the uptime field will be measured in **seconds**.

### CODE REVIEW

**Comment by elasticcla** [comment]
Hi @mdelapenya, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/ecs/pull/477.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by mdelapenya** [comment]
Hey @webmat, I updated the PR to include unit suffix as first-class citizen, so the new fields are
`process.uptime.seconds` and `server.uptime.seconds`

cc/ @axw

**Comment by webmat** [comment]
@mdelapenya Thanks for updating the PR. My suggestion was to make it seconds, but I should have been more explicit: the unit shouldn't be part of the field name. We should have the description state that it's seconds.

In the context of the ecs-metrics experimental branch, we're even considering making this directive official, see https://github.com/elastic/ecs/pull/481

**Comment by mdelapenya** [comment]
Oh my! I understood totally the opposite from the thread :) Also reading [how prometheus folks define metrics](https://prometheus.io/docs/practices/naming/#metric-names) made me think that way.

Anyway, I'll revert back to the original, where units were explicitly described in fields comments/descriptions

Thanks!

**Comment by mdelapenya** [comment]
PR updated with your suggestions, thanks!

File: schemas/server.yml [hunk 1]
```diff
@@ -74,3 +74,10 @@
       example: 12
       description: >
         Packets sent from the server to the client.
+
+    - name: uptime
```
Comment (webmat) [inline schemas/server.yml]: Sorry I hadn't noticed this. The field set `server` is meant to descrive the server side of a network connection. We have two pairs: client/server and source/destination. As such, this isn't the right place for this.

`server.uptime` should be moved to `host.uptime` :-)

Let's also make `host.uptime` extended for now.

File: schemas/process.yml [hunk 2]
```diff
@@ -87,6 +87,13 @@
       description: >
         The time the process started.
 
+    - name: uptime
+      level: core
```
Comment (webmat) [inline schemas/process.yml]: Please make this `extended` for now.

File: schemas/server.yml (hunk 1 above)
Comment (mdelapenya) [reply schemas/server.yml]: Is `extended` a matter of adding a `level: extended`, or is there any other consideration?

**Comment by mdelapenya** [comment]
Changes ready!

**Comment by mdelapenya** [comment]
Thanks @webmat!! What about the docs? Is this something I have to handle too?

**Comment by ruflin** [comment]
@mdelapenya Docs were generated automatically.

**Comment by webmat** [comment]
Yes, generated every half hour. But note that master != current on the docs. So you can see your contribution if you go to the docs and select "master" in the version selector.

**Pull request merged**